src/vtlengine/Operators/Time.py

```diff
@@ -437,10 +439,8 @@ def evaluate(cls, operand: Dataset, shift_value: Scalar) -> Dataset:
         data_type = result.components[cls.time_id].data_type
 
         if data_type == Date:
-            freq = cls.find_min_frequency(
-                cls.get_frequencies(result.data[cls.time_id].apply(cls.parse_date)))
-            result.data[cls.time_id] = result.data[cls.time_id].apply(
-                lambda x: cls.shift_date(x, shift_value, freq)).astype(str)
+            freq = cls.find_min_frequency(cls.get_frequencies(result.data[cls.time_id].apply(cls.parse_date)))
```

Comment:
Use map here with na_action='ignore' to enhance performance and prevent null issues, instead of apply